pandas/tests/test_window.py

```diff
@@ -207,6 +207,38 @@ def f():
             'A', 'ra', 'std'), ('B', 'rb', 'mean'), ('B', 'rb', 'std')])
         tm.assert_frame_equal(result, expected, check_like=True)
 
+    def test_count_nonnumeric_types(self):
+        # GH12541
+        cols = ['int', 'float', 'string', 'datetime', 'timedelta',
+                'fl_inf', 'fl_nan', 'str_nan', 'dt_nat']
+
+        df = DataFrame(
+            {'int': [1, 2, 3],
+             'float': [4., 5., 6.],
+             'string': list('abc'),
+             'datetime': pd.date_range('20170101', periods=3),
+             'timedelta': pd.timedelta_range('1 s', periods=3, freq='s'),
+             'fl_inf': [1., 2., np.Inf],
+             'fl_nan': [1., 2., np.NaN],
+             'str_nan': ['aa', 'bb', np.NaN],
+             'dt_nat': [pd.Timestamp('20170101'), pd.Timestamp('20170203'),
+                        pd.Timestamp(None)]},
```

Comment:
Something like this?

`df = DataFrame(`
`            {...,`
`            'periods': [pd.Period('2012-01'), pd.Period('2012-02'), `
`                       pd.Period('2012-03')],`
`             ...`
`             'periods_nat': [pd.Period('2012-01'), pd.Period('2012-02'), `
`                         pd.Period(None)]},`
`            columns=cols)`